libr/socket/socket.c

```diff
@@ -731,7 +743,9 @@ R_API char *r_socket_to_string(RSocket *s) {
 R_API int r_socket_write(RSocket *s, void *buf, int len) {
 	int ret, delta = 0;
 #if __UNIX__ || defined(__CYGWIN__)
-	signal (SIGPIPE, SIG_IGN);
+	sigset_t mask;
+	sigemptyset (&mask);
+	r_sys_sigaction (SIGPIPE, SIG_IGN, &mask, 0);
```

Comment:
imho the r_sys_sigaction api shuold return the mask, initialized.

```c
RSysSigAction mask = r_sys_sigaction (SIGPIPE, SIG_IGN, 0, -1);
```

this way you can safe 3 lines, and put all the logic inside r_sys_sigaction

Reply:
* Why would it return the signal mask, we are not gonna need it.

* Please read my last comment:

> You are forgetting that choosing the signal mask is separate from choosing for which signal to install a handler (that is among other things why sigaction needs to replace signal).